EIPS/eip-7525.md

```diff
@@ -0,0 +1,161 @@
+
+### Account State Change
+
+When an account is expired, the nonce value of that address is actually reset to 0. When the nonce is reset, it can no longer serve to prevent replay attacks. To prevent such issues, we additionally store in the account state the `restored_epoch`, which is information about the epoch when it was restored. By checking not only the nonce but also the `restored_epoch` when executing a transaction, we prevent old transactions from being executed again.
+
+The `restored_epoch` not only plays a role similar to the nonce during the restoration process, but also makes it possible to selectively determine the point of restoration, eliminating the need to verify the state from the genesis block. This also resolves cost or size issues by allowing the restoration to be carried out over multiple transactions.
+
+Additionally, it plays a crucial role in contract creation, helping to avoid regeneration of the same address.
+
+### Restoration Process
+
+```python
+def restore_account(account, proofs):
+	restored_epoch = account.restored_epoch
+
+  for proof in proofs:
+		root_hash = get_last_checkpoint_block_by_epoch(restored_epoch).state_root
+
+		if is_accurate_merkle_proof(root_hash, account, proof): # Proof is non-void proof
+        restored_account = extract_account(merkle_proof)
+        account.restored_epoch = restored_account.restored_epoch
+        account.balance += restored_account.balance
+				account.nonce += restored_account.nonce
+    elif is_accurate_void_proof(root_hash, account, proof): # Proof is void proof
+        restored_epoch -= 1
+    else: # Proof is invalid
+        raise Exception("Inaccurate proof")
+```
+
+### Restoration Cost
+
+| Operation | Cost (Gwei) |
+| --- | --- |
+| read restoredEpoch | 20 |
+| read nonce | 20 |
+| read balance | 20 |
+| Keccak256 | 100 |
+| Ecrecover | 3000 |
+| CallValueTransfer | 9000 |
+| CallNewAccount | 25000 |
+| read header for every epoch | dynamic |
+| RLP decoding on every non-void epoch | dynamic |
+| verifyProof for every epoch | dynamic |
+| RLP decoding on input data | dynamic |
+
+Dynamic cost refers to the cost that varies with each epoch and is proportional to the length of the input.
+
+The cost incurred in each epoch consists of 1 **`verifyProof`** and 1 **`read header`** operation. If the epoch is non-void, it also includes 1 **`RLP decoding`** operation. These operations incur a cost of approximately 900 Gwei per epoch.
+
+The cost that varies with the length of the input consists of 1 **`RLP decoding`** and 1 **`verifyProof`** operation, both proportional to the size of the input. These operations incur a cost of 3 Gwei per word.
+
+The total recovery cost is as follows:
+
+$$
+\text{Total Restoration Cost} = 37000 + 900\times{Epoch} + 3\times{words} + \text{Memory Cost}
+$$
+
+## Rationale
+
+The rationale behind this EIP is anchored on enhancing network efficiency, scalability, and accessibility to the network.
```

Comment:
rationale is about your approach, why you took this approach its pros vs the others

Reply:
like why you made certain design decisions like setting `nonce` to zero etc, not restoring storage....